makefile

```diff
@@ -74,28 +87,40 @@ FINUFFT = $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
 # -fPIC (position-indep code) needed to build dyn lib (.so)
 # Also, we force return (via :=) to the land of simply-expanded variables...
 INCL = -Iinclude -I$(XSIMD_DIR)/include
+# single-thread total list of math and FFT libs (now both precisions)...
+# (Note: finufft tests use LIBSFFT; spread & util tests only need LIBS)
+LIBSFFT := $(LIBS)
+ifeq ($(FFT),DUCC)
+  DUCC_SETUP := $(DUCC_COOKIE)
+# so FINUFFT build can see DUCC headers...
+  INCL += $(DUCC_INCL)
+  DUCC_OBJS := $(DUCC_SRC)/infra/string_utils.o $(DUCC_SRC)/infra/threading.o $(DUCC_SRC)/infra/mav.o $(DUCC_SRC)/math/gridding_kernel.o $(DUCC_SRC)/math/gl_integrator.o
+# FINUFFT's switchable FFT done via this compile directive...
+  CXXFLAGS += -DFINUFFT_USE_DUCC0
+else
+# link against FFTW3 single-threaded (leaves DUCC_OBJS and DUCC_SETUP undef)
+  LIBSFFT += -l$(FFTWNAME) -l$(FFTWNAME)f
+endif
 CXXFLAGS := $(CXXFLAGS) $(INCL) -fPIC -std=c++17
 CFLAGS := $(CFLAGS) $(INCL) -fPIC
 # here /usr/include needed for fftw3.f "fortran header"... (JiriK: no longer)
 FFLAGS := $(FFLAGS) $(INCL) -I/usr/include -fPIC
 
-# single-thread total list of math and FFTW libs (now both precisions)...
-# (Note: finufft tests use LIBSFFT; spread & util tests only need LIBS)
-LIBSFFT := -l$(FFTWNAME) -l$(FFTWNAME)f $(LIBS)
-
 # multi-threaded libs & flags, and req'd flags (OO for new interface)...
 ifneq ($(OMP),OFF)
   CXXFLAGS += $(OMPFLAGS)
   CFLAGS += $(OMPFLAGS)
   FFLAGS += $(OMPFLAGS)
-  MFLAGS += $(MOMPFLAGS) -DR2008OO
-  OFLAGS += $(OOMPFLAGS) -DR2008OO
+  MFLAGS += $(MOMPFLAGS)
+  OFLAGS += $(OOMPFLAGS)
   LIBS += $(OMPLIBS)
-# omp override for total list of math and FFTW libs (now both precisions)...
-  LIBSFFT := -l$(FFTWNAME) -l$(FFTWNAME)_$(FFTWOMPSUFFIX) -l$(FFTWNAME)f -l$(FFTWNAME)f_$(FFTWOMPSUFFIX) $(LIBS)
+# fftw3 multithreaded libs...
+  ifneq ($(FFT),DUCC)
+    LIBSFFT += -l$(FFTWNAME)_$(FFTWOMPSUFFIX) -l$(FFTWNAME)f_$(FFTWOMPSUFFIX)
```

Comment:
should we add back `$(LIBS)` to `LIBSFFT`, i.e.
`LIBSFFT += -l$(FFTWNAME)_$(FFTWOMPSUFFIX) -l$(FFTWNAME)f_$(FFTWOMPSUFFIX) $(LIBS)`
In the above comment "# (Note: finufft tests use LIBSFFT; spread & util tests only need LIBS)"
LIBSFFT contains the omp lib to link, without omp lib in LIBSFFT, the ci for mac clang breaks for not finding omp lib to link.

Reply:
seems CI pass now